File: excelium-executor/src/main/java/excelium/executor/DateCommandExecutor.java
/*
 * MIT License
 *
 * Copyright (c) 2018 Excelium
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package excelium.executor;

import excelium.core.command.Accessor;
import excelium.core.driver.ContextAwareWebDriver;
import excelium.core.Excelium;
import excelium.core.CommandExecutor;
import excelium.model.project.Project;

import java.util.Calendar;
import java.util.Date;

/**
 * Represents a class which contains commands for controlling dates.
 *
 * @author PhungDucKien
 * @since 2018.07.05
 */
public class DateCommandExecutor extends CommandExecutor {

    /**
     * Instantiates a new Date command executor.
     *
     * @param webDriver the web driver
     * @param baseUrl   the base url
     * @param excelium  the excelium
     * @param project   the project
     */
    public DateCommandExecutor(ContextAwareWebDriver webDriver, String baseUrl, Excelium excelium, Project project) {
        super(webDriver, baseUrl, excelium, project);
    }

    /**
     * Returns the current date.
     *
     * @return the current date
     */
    @Accessor(verifyCmd = false, waitCmd = false, executeCmd = false)
    public Date getCurrentDate() {
        return Calendar.getInstance().getTime();
    }
}

File: packages/executor/src/DateCommandExecutor.ts
/*
 * Licensed to the Software Freedom Conservancy (SFC) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The SFC licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { Project } from '@excelium/model';
import { Accessor, Action, CommandExecutor, WebDriverExecutor } from '@excelium/runtime';

/**
 * Represents a class which contains commands for controlling dates.
 *
 * @author PhungDucKien
 * @since 2018.07.05
 */
export default class DateCommandExecutor extends CommandExecutor {
  /**
   * Instantiates a new Date command executor.
   *
   * @param webDriver the web driver
   * @param project   the project
   */
  public constructor(webDriver: WebDriverExecutor, project: Project) {
    super(webDriver, project);
  }

  public getActions(): Array<Action<any>> {
    return [
      new Accessor({
        execute: this.getCurrentDate.bind(this),
        exclude: ['verify', 'wait', 'execute'],
      }),
    ];
  }

  /**
   * Returns the current date.
   *
   * @return the current date
   */
  public async getCurrentDate() {
    return new Date();
  }
}
